manifests/01-validating-admission-policy.yaml

```diff
@@ -0,0 +1,36 @@
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: openshift-ingress-operator-gatewayapi-crd-admission
+  annotations:
+    capability.openshift.io/name: Ingress
+    include.release.openshift.io/ibm-cloud-managed: "true"
+    include.release.openshift.io/self-managed-high-availability: "true"
+    include.release.openshift.io/single-node-developer: "true"
+    release.openshift.io/feature-set: "TechPreviewNoUpgrade,CustomNoUpgrade"
+spec:
+  matchConstraints:
```

Comment:
The godocs on matchConstraints are not very good.  

> // The MutatingAdmissionPolicy cares about a request if it matches _all_ Constraint.
> // However, in order to prevent clusters from being put into an unstable state that cannot be recovered from via the API
> // ValidatingAdmissionPolicy cannot match ValidatingAdmissionPolicy and ValidatingAdmissionPolicyBinding.

I looked into the Kubernetes reference and found in https://kubernetes.io/docs/reference/access-authn-authz/validating-admission-policy/#type-checking, this:

> Type Checking does not apply to CRDs, including matched CRD types and reference of paramKind. The support for CRDs will come in future release.

Does it apply to us?  No issues with adding this VAP?

Reply:
I think that "matched CRD types" refers to the type that is defined by the CRD, not to the CRD type itself.  That is, I expect that the VAP can block updates to, say, the gatewayclasses CRD, but it cannot at this time block updates to an arbitrary gatewayclass CR.

Reply:
> Does it apply to us? No issues with adding this VAP?

The type checking will be missing for this VAP. Meaning that when the VAP will be updated and if it will contain any syntax mistakes in validation expressions (or variables) the update of VAP will be blocked and the syntax error will be reported in the VAP's status. I don't think it's a blocking gap because we will validate our VAP before it will hit any cluster (local testing, integration testing, QE testing).

Reply:
> The godocs on matchConstraints are not very good.
> 
> >     // The MutatingAdmissionPolicy cares about a request if it matches all Constraint.
> >     // However, in order to prevent clusters from being put into an unstable state that cannot be recovered from via the API
> >     // ValidatingAdmissionPolicy cannot match ValidatingAdmissionPolicy and ValidatingAdmissionPolicyBinding.

You mean the part about MutatingAdmissionPolicy? Indeed, it's unclear but should not impact us in anyway as we are adding VAP.
The otherpart about the ValidatingAdmissionPolicy is a known constraint which does not impact us either because we don't need to match VAPs or VAP bindings.

Reply:
What does that other part even mean?  I can't understand the grammar.

Reply:
> // ValidatingAdmissionPolicy cannot match ValidatingAdmissionPolicy and ValidatingAdmissionPolicyBinding.

This part means that it's not possible to create ValidatingAdmissionPolicy which validates the admission of another ValidatingAdmissionPolicy or ValidatingAdmissionPolicyBinding.